app/controllers/api/v1/freshdesk/ticket_controller.rb

```diff
@@ -0,0 +1,265 @@
+  before_action :check_user_defaults
+  before_action :check_user_email
+  before_action :httparty_default_setting
+  rescue_from StandardError, :with => :catch_error
+
+  def index
+    verify_params(params, [:page_no, :order_by])
+    all_tickets_res = self.class.get("/tickets?email=#{@email}&order_by=#{params[:order_by]}&per_page=#{@tickets_per_request}&page=#{params[:page_no]}")
+    validate_response(all_tickets_res)
+    all_tickets_res = JSON.parse(all_tickets_res.body)
+    open_tickets = all_tickets_res.select { |ticket| [2,3].include?(ticket["status"]) }
+    close_tickets = all_tickets_res.select { |ticket| [4,5].include?(ticket["status"]) }
+    render json: { :open => open_tickets, :close => close_tickets }, status: 200
+  end
+
+  def init_settings
+    render json: { :per_page => @per_page, :route => Freshdesk.routes, :tickets_per_request => @tickets_per_request }, status: 200
+  end
+
+  def read
+    verify_params(params, [:id, :user_id])
+    ticket_res = self.class.get("/tickets/#{params[:id]}")
+    validate_response(ticket_res)
+    ticket_res = JSON.parse(ticket_res.body)
+    raise BlogVault::NotFoundError.new('Ticket') if (ticket_res["requester_id"] != params[:user_id])
+    conversation_res = self.class.get("/tickets/#{params[:id]}/conversations")
+    validate_response(conversation_res)
+    conversation_res = JSON.parse(conversation_res.body)
+    ticket_res["conversationList"] = conversation_res.select{|conversation| conversation['private'] == false}
+    render json: ticket_res, status: 200
+  end
+
+  def new
+    ticket_fields_res = fetch_ticket_fields
+    ticket_fields_res = ticket_field_filter(ticket_fields_res)
+    render json: ticket_fields_res, status: 200
+  end
+
+  def create
+    required_fields = get_required_fields(fetch_ticket_fields)
+    verify_params(params, required_fields)
+    contact_exists?
+    body = required_field(params, [:attachments, :subject, :description, :custom_fields])
+    body[:email] = @email
+    body[:priority] = 1
+    body[:status] = 2	
+    res = self.class.post('/tickets', {
+      :body => body,
+      :headers => {"Content-Type" => 'multipart/form-data'} }
+    )
+    validate_response(res)
+    render json: res.body, status: res.code
+  end
+
+  def update
+    verify_params(params, [:id, :status])
+    res = self.class.put("/tickets/#{params[:id]}",{:body => { status: params[:status]}.to_json(),:headers => {"Content-Type" => "application/json"}})
+    validate_response(res)
+    render json: res.body, status: res.code
+  end
+
+  def reply
+    verify_params(params, [:agent_id, :body, :id, :user_id])
+    body = required_field(params, [:body, :user_id, :attachments])
+    body[:private] = false
+    if params[:agent_id] != "null"
+      agent_res = self.class.get("/agents/#{params[:agent_id]}")
+      validate_response(agent_res)
+      agent_res = JSON.parse(agent_res.body)
+      body[:notify_emails] =  [agent_res["contact"]["email"]]
+    end
```

Comment:
Do agent can reply here

Reply:
It is checking whether there is any agent associated to a ticket and then it will notify that agent about the comment that user made in the ticket.